public/app/features/explore/state/query.ts

```diff
@@ -608,17 +636,17 @@ export const runQueries = (
  * Checks if after changing the time range the existing data can be used to show logs volume.
  * It can happen if queries are the same and new time range is within existing data time range.
  */
-function canReuseLogsVolumeData(
-  logsVolumeData: DataQueryResponse | undefined,
+function canReuseSuppQueryData(
+  suppQueryData: DataQueryResponse | undefined,
   queries: DataQuery[],
   selectedTimeRange: AbsoluteTimeRange
 ): boolean {
-  if (logsVolumeData && logsVolumeData.data[0]) {
+  if (suppQueryData && suppQueryData.data[0]) {
     // check if queries are the same
-    if (!deepEqual(logsVolumeData.data[0].meta?.custom?.targets, queries)) {
+    if (!deepEqual(suppQueryData.data[0].meta?.custom?.targets, queries)) {
       return false;
     }
-    const dataRange = logsVolumeData && logsVolumeData.data[0] && logsVolumeData.data[0].meta?.custom?.absoluteRange;
+    const dataRange = suppQueryData && suppQueryData.data[0] && suppQueryData.data[0].meta?.custom?.absoluteRange;
```

Comment:
`suppQueryData && suppQueryData.data[0]` shouldn't be necessary anymore here.

Reply:
It is, because suppQueryData can be  `DataQueryResponse | undefined`.

Reply:
Yep, but you already check for that in line 644: `if (suppQueryData && suppQueryData.data[0]) {`